### PR TITLE
[REG-1449] feat: Update ENSCustody contract to support wallet migration

### PR DESCRIPTION
**Pull request by DancingAxolotl**
Related: #369 

### 1. Contracts versioning
- [x] Make sure that the `patch` version of the contracts is increased if changes have been made to the `UNSRegistry`, `MintingManager`, `ProxyReader`, `ENSCustody` contracts.
- [ ] Make sure that the `minor` version of the contracts is increased if breaking changes have been made to the `UNSRegistry`, `MintingManager`, `ProxyReader`, `ENSCustody` contracts. It includes changes of interfaces.
### 2. Contracts licensing
- [ ] Make sure that no **SPDX-License-Identifier** defined in contracts.
- [ ] Make sure that the **header** is added to the new contract files.
  ```
  // @author Unstoppable Domains, Inc.
  // @date {Month} {Day}(ordinal), {Year}
  ```
### 3. Coverage
- [ ] Make sure that the coverage of contracts has not decreased and strive **100%**
### 4. Configs versioning
- [ ] Make sure that the version of `uns-config.json` is increased if changes have been made to the config.
- [ ] Make sure that the version of `ens-config.json` is increased if changes have been made to the config.
- [ ] Make sure that the version of `resolver-keys.json` is increased if changes have been made to the config.
- [ ] Make sure that the version of `ens-resolver-keys.json` is increased if changes have been made to the config.
### 5. Package versioning
- [x] Make sure that the `patch` version of package is increased if valuable changes have been made to the package. It includes contracts update, configs update, etc.
- [ ] Make sure that the `major.minor` version of package is synced with version of `UNSRegistry` contract.
- [ ] Make sure that the `CHANGELOG` is updated with short description for the new version.
### 6. Code review
- [ ] `resolver-keys.json` code review is required from **DevTools** team
- [ ] `ens-resolver-keys.json` code review is required from **DevTools** team
- [ ] For all other changes code review is required from **Registry** team

### CODE REVIEW

File: test/custody/ENSCustody.test.ts [hunk 1]
```diff
@@ -957,6 +957,199 @@ describe('ENSCustody', function () {
...
+      await registerAndParkName(name, minter, ZERO_ADDRESS, false);
+
+      await expect(custody.connect(minter).internalTransfer(newOwnerAddress, tokenId)).to.be.revertedWithCustomError(
+        custody,
+        'Unauthorised',
+      );
+    });
+  });
+
+  describe('multicall', async () => {
```
Comment (DeRain) [inline test/custody/ENSCustody.test.ts:1067]: Maybe one more test to validate that ownership model is not broken via multi call. Like revert if not valid signer.

File: test/custody/ENSCustody.metatx.test.ts [hunk 2]
```diff
@@ -285,4 +330,139 @@ describe('ENSCustody (metatx)', function () {
       'InvalidForwardedToken',
     );
   });
+
+  describe('Multicall', () => {
+    it('should execute meta tx with multicall', async () => {
```
Comment (DeRain) [inline test/custody/ENSCustody.metatx.test.ts:335]: Same here. One more test to check ownership model and proper reverts in case of incorrect signer

**Review by DeRain — APPROVED** [review]

File: test/custody/ENSCustody.test.ts (hunk 1 above)
Comment (nickshatilo) [reply test/custody/ENSCustody.test.ts:1067]: for this particular one we already have a test, see the second one

File: test/custody/ENSCustody.metatx.test.ts (hunk 2 above)
Comment (nickshatilo) [reply test/custody/ENSCustody.metatx.test.ts:336]: added

**Pull request merged**